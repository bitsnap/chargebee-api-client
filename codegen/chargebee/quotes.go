package chargebee

import (
	"github.com/bitsnap/chargebee-api-client/codegen/client"
)

// GenerateListQuotes generates chargebee client code to fetch all quotes
//
// API: https://{site}.chargebee.com/api/v2/quotes
//
// Documentation: https://apidocs.chargebee.com/docs/api/quotes?lang=curl#list_quotes
func GenerateListQuotes() string {
	return client.GenerateList("Quotes", "Quote", "chargebee.com/api/v2/quotes")
}

// GenerateRetrieveQuote generates chargebee client code to retrieve specific quote
//
// API: https://{site}.chargebee.com/api/v2/quotes/{quote-id}
//
// Documentation: https://apidocs.chargebee.com/docs/api/quotes?lang=curl#retrieve_a_quote
func GenerateRetrieveQuote() string {
	return client.GenerateRetrieve("Quote", "chargebee.com/api/v2/quotes")
}

// GenerateQuotesModel generates chargebee Quote domain model
//
// Documentation: https://apidocs.chargebee.com/docs/api/quotes?lang=curl#quote_attributes
func GenerateQuotesModel() string {
	return client.GenerateModel("Quote", "Quote attributes", "https://apidocs.chargebee.com/docs/api/quotes")
}
