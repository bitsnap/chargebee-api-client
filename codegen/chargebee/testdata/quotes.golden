# Generated by goldga. DO NOT EDIT.
[snapshots]
"Chargebee quotes code generation when generating Chargebee API client should be able to generate quotes models" = '''
(string) (len=6256) "type Quote struct {\n    Id string `json:\"id\" validate:\"required\"`\n    Name string `json:\"name\"`\n    PoNumber string `json:\"po_number\"`\n    CustomerId string `json:\"customer_id\" validate:\"required\"`\n    SubscriptionId string `json:\"subscription_id\"`\n    InvoiceId string `json:\"invoice_id\"`\n    Status enums.StatusEnum `json:\"status\" validate:\"required\"`\n    OperationType enums.OperationTypeEnum `json:\"operation_type\" validate:\"required\"`\n    VatNumber string `json:\"vat_number\"`\n    PriceType enums.PriceTypeEnum `json:\"price_type\" validate:\"required\"`\n    ValidTill *time.Time `json:\"valid_till\" validate:\"required\"`\n    Date *time.Time `json:\"date\" validate:\"required\"`\n    TotalPayable uint64 `json:\"total_payable\"`\n    ChargeOnAcceptance uint64 `json:\"charge_on_acceptance\"`\n    SubTotal uint64 `json:\"sub_total\" validate:\"required\"`\n    Total uint64 `json:\"total\"`\n    CreditsApplied uint64 `json:\"credits_applied\"`\n    AmountPaid uint64 `json:\"amount_paid\"`\n    AmountDue uint64 `json:\"amount_due\"`\n    Version int `json:\"version\"`\n    ResourceVersion int64 `json:\"resource_version\"`\n    UpdatedAt *time.Time `json:\"updated_at\"`\n    VatNumberPrefix string `json:\"vat_number_prefix\"`\n    TaxCategory string `json:\"tax_category\"`\n    CurrencyCode string `json:\"currency_code\" validate:\"required\"`\n    ContractTermStart *time.Time `json:\"contract_term_start\"`\n    ContractTermEnd *time.Time `json:\"contract_term_end\"`\n    ContractTermTerminationFee uint64 `json:\"contract_term_termination_fee\"`\n    BusinessEntityId string `json:\"business_entity_id\"`\n    LineItems []LineItem `json:\"line_items\"`\n    Id string `json:\"id\"`\n    SubscriptionId string `json:\"subscription_id\"`\n    DateFrom *time.Time `json:\"date_from\" validate:\"required\"`\n    DateTo *time.Time `json:\"date_to\" validate:\"required\"`\n    UnitAmount uint64 `json:\"unit_amount\" validate:\"required\"`\n    Quantity int `json:\"quantity\"`\n    Amount uint64 `json:\"amount\"`\n    PricingModel enums.PricingModelEnum `json:\"pricing_model\"`\n    IsTaxed bool `json:\"is_taxed\" validate:\"required\"`\n    TaxAmount uint64 `json:\"tax_amount\"`\n    UnitAmountInDecimal string `json:\"unit_amount_in_decimal\"`\n    QuantityInDecimal string `json:\"quantity_in_decimal\"`\n    AmountInDecimal string `json:\"amount_in_decimal\"`\n    DiscountAmount uint64 `json:\"discount_amount\"`\n    ItemLevelDiscountAmount uint64 `json:\"item_level_discount_amount\"`\n    ReferenceLineItemId string `json:\"reference_line_item_id\"`\n    Description string `json:\"description\" validate:\"required\"`\n    EntityDescription string `json:\"entity_description\"`\n    EntityType enums.EntityTypeEnum `json:\"entity_type\" validate:\"required\"`\n    TaxExemptReason enums.TaxExemptReasonEnum `json:\"tax_exempt_reason\"`\n    EntityId string `json:\"entity_id\"`\n    CustomerId string `json:\"customer_id\"`\n    Discounts []Discount `json:\"discounts\"`\n    Amount uint64 `json:\"amount\" validate:\"required\"`\n    Description string `json:\"description\"`\n    LineItemId string `json:\"line_item_id\"`\n    EntityType enums.EntityTypeEnum `json:\"entity_type\" validate:\"required\"`\n    DiscountType enums.DiscountTypeEnum `json:\"discount_type\"`\n    EntityId string `json:\"entity_id\"`\n    CouponSetCode string `json:\"coupon_set_code\"`\n    LineItemDiscounts []LineItemDiscount `json:\"line_item_discounts\"`\n    LineItemId string `json:\"line_item_id\" validate:\"required\"`\n    DiscountType enums.DiscountTypeEnum `json:\"discount_type\" validate:\"required\"`\n    EntityId string `json:\"entity_id\"`\n    DiscountAmount uint64 `json:\"discount_amount\" validate:\"required\"`\n    Taxes []Tax `json:\"taxes\"`\n    Name string `json:\"name\" validate:\"required\"`\n    Amount uint64 `json:\"amount\" validate:\"required\"`\n    Description string `json:\"description\"`\n    LineItemTaxes []LineItemTax `json:\"line_item_taxes\"`\n    LineItemId string `json:\"line_item_id\"`\n    TaxName string `json:\"tax_name\" validate:\"required\"`\n    DateTo *time.Time `json:\"date_to\"`\n    DateFrom *time.Time `json:\"date_from\"`\n    ProratedTaxableAmount *big.Float `json:\"prorated_taxable_amount\"`\n    IsPartialTaxApplied bool `json:\"is_partial_tax_applied\"`\n    IsNonComplianceTax bool `json:\"is_non_compliance_tax\"`\n    TaxableAmount uint64 `json:\"taxable_amount\" validate:\"required\"`\n    TaxAmount uint64 `json:\"tax_amount\" validate:\"required\"`\n    TaxJurisType enums.TaxJurisTypeEnum `json:\"tax_juris_type\"`\n    TaxJurisName string `json:\"tax_juris_name\"`\n    TaxJurisCode string `json:\"tax_juris_code\"`\n    TaxAmountInLocalCurrency uint64 `json:\"tax_amount_in_local_currency\"`\n    LocalCurrencyCode string `json:\"local_currency_code\"`\n    LineItemTiers []LineItemTier `json:\"line_item_tiers\"`\n    LineItemId string `json:\"line_item_id\"`\n    StartingUnit int `json:\"starting_unit\" validate:\"required\"`\n    EndingUnit int `json:\"ending_unit\"`\n    QuantityUsed int `json:\"quantity_used\" validate:\"required\"`\n    UnitAmount uint64 `json:\"unit_amount\" validate:\"required\"`\n    StartingUnitInDecimal string `json:\"starting_unit_in_decimal\"`\n    EndingUnitInDecimal string `json:\"ending_unit_in_decimal\"`\n    QuantityUsedInDecimal string `json:\"quantity_used_in_decimal\"`\n    UnitAmountInDecimal string `json:\"unit_amount_in_decimal\"`\n    FirstName string `json:\"first_name\"`\n    LastName string `json:\"last_name\"`\n    Email string `json:\"email\"`\n    Company string `json:\"company\"`\n    Phone string `json:\"phone\"`\n    Line1 string `json:\"line1\"`\n    Line2 string `json:\"line2\"`\n    Line3 string `json:\"line3\"`\n    City string `json:\"city\"`\n    StateCode string `json:\"state_code\"`\n    State string `json:\"state\"`\n    Country string `json:\"country\"`\n    Zip string `json:\"zip\"`\n    ValidationStatus enums.ValidationStatusEnum `json:\"validation_status\"`\n    Index int `json:\"index\" validate:\"required\"`\n    FirstName string `json:\"first_name\"`\n    LastName string `json:\"last_name\"`\n    Email string `json:\"email\"`\n    Company string `json:\"company\"`\n    Phone string `json:\"phone\"`\n    Line1 string `json:\"line1\"`\n    Line2 string `json:\"line2\"`\n    Line3 string `json:\"line3\"`\n    City string `json:\"city\"`\n    StateCode string `json:\"state_code\"`\n    State string `json:\"state\"`\n    Country string `json:\"country\"`\n    Zip string `json:\"zip\"`\n    ValidationStatus enums.ValidationStatusEnum `json:\"validation_status\"`\n}"
'''
